### ISSUE TITLE
Memory leak in mido.get_input_names() on Windows 10 with python-rtmidi

### ISSUE THREAD

**Issue by eshapiro42** [mido/mido#236]
I was trying to work on a solution to issue #224 by detecting a disconnect event and then polling the output of `mido.get_input_names()` until the desired device reappeared. My solution worked, but when I left it running overnight I woke up to my program consuming over 4GB of memory.

I traced it back to this function call. The following is some minimal runnable code which demonstrates the problem:

```py
import mido
import time

while True:
    input_names = mido.get_input_names()
    time.sleep(0.01)
```

This quickly consumes increasingly more memory the longer it is run.

I am running Python 3.8.4 on Windows 10 with the python-rtmidi backend.

It should be noted that the issue is _not_ with python-rtmidi. The following code has no memory leaks:

```py
import rtmidi
import time

midi_in = rtmidi.MidiIn()
while True:
    input_names = midi_in.get_ports()
    time.sleep(0.01)
````

**Comment by goc9000** [comment]
This is also happening for me with Python 3.8.5 on OS X.

**Comment by olemb** [comment]
Thanks for the report!

It looks like @TheOtherDays has a fix for this in issue #256.

**Comment by rdoursenaud** [comment]
Since #256 has been merged, I guess its fixed.
Feel free to reopen if not.
Thanks!
